src/Spectre.libClassifier/OpenCvDataset.cpp

```diff
@@ -48,6 +48,30 @@ OpenCvDataset::OpenCvDataset(OpenCvDataset &&other) noexcept
     other.m_observations.clear();
 }
 
+OpenCvDataset::OpenCvDataset(OpenCvDataset &&first, OpenCvDataset &&second) noexcept
+    : m_Data(std::move(first.m_Data)),
+    m_labels(std::move(first.m_labels)),
+    m_observations(std::move(first.m_observations))
+{
+    m_Data.insert(m_Data.end(), second.m_Data.begin(), second.m_Data.end());
+    m_labels.insert(m_labels.end(), second.m_labels.begin(), second.m_labels.end());
+    m_observations.insert(m_observations.end(), second.m_observations.begin(), second.m_observations.end());
+
+    m_Mat = cv::Mat(static_cast<int>(m_labels.size()), static_cast<int>(m_Data.size() / m_labels.size()), CV_TYPE, m_Data.data());
+    m_MatLabels = cv::Mat(static_cast<int>(m_labels.size()), ColumnMatrixWidth, CV_LABEL_TYPE, m_labels.data());
+
+    first.m_Data.clear();
```

Comment:
If there's so much cleanup to do, extract it to a `clear` function.

Reply:
done